app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,634 @@
+            return searchData;
+        }
+
+        return (searchData as DialogOption[]).filter((option) => option.text && option.text.toLowerCase().startsWith(lowerCasedTerm));
+    };
+
+    const getMultiselectData = useCallback((): Selection => {
+        let myItems;
+        let multiselectItems: Selection = [];
+
+        switch (dataSource) {
+            case ViewConstants.DATA_SOURCE_USERS:
+                myItems = multiselectSelected as Dictionary<UserProfile>;
+                multiselectItems = multiselectItems as UserProfile[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+            case ViewConstants.DATA_SOURCE_CHANNELS:
+                myItems = multiselectSelected as Dictionary<Channel>;
+                multiselectItems = multiselectItems as Channel[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+            default:
+                myItems = multiselectSelected as Dictionary<DialogOption>;
+                multiselectItems = multiselectItems as DialogOption[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+        }
+    }, [multiselectSelected]);
+
+    // Effects
+    useNavButtonPressed(SUBMIT_BUTTON_ID, componentId, onHandleMultiselectSubmit, [onHandleMultiselectSubmit]);
+
+    useEffect(() => {
+        setLoading(true);
+
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            getProfiles();
+        } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+            getChannels();
+        } else {
+            getDynamicOptionsLocally();
+        }
+
+        setLoading(false);
+    }, []);
+
+    useEffect(() => {
+        let listData: (DataType | UserProfileSection[]) = integrationData;
+        if (term) {
+            listData = searchResults;
+        }
+
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            listData = createProfilesSections(listData as UserProfile[]);
+        }
+
+        if (!dataSource || dataSource === ViewConstants.DATA_SOURCE_DYNAMIC) {
+            listData = (integrationData as DialogOption[]).filter((option) => option.text && option.text.toLowerCase().includes(term));
+        }
+
+        setCustomListData(listData);
+    }, [searchResults, integrationData]);
+
+    useEffect(() => {
+        if (!isMultiselect) {
+            return;
+        }
+
+        setButtons(componentId, {
+            rightButtons: [rightButton],
+        });
+    }, [rightButton, componentId]);
```

Comment:
0/5 for future proof, you can add `isMultiselect` to the dependencies, even if we know that it won't change during the lifetime of the component.